lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx

```diff
@@ -4,207 +4,309 @@
+        }),
+        dropdownToggleBtn = screen.getByRole('button', { name: 'more options' });
+
+    await user.click(dropdownToggleBtn);
+    let menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await act(async () => { await user.click(screen.getByTestId('child')); });
     expect(childClickSpy).toHaveBeenCalled();
-    expect(element!.find('.nx-dropdown-menu')).not.toExist();
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
 
-    act(() => {
-      element!.find('button.nx-segmented-btn__dropdown-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
+    await user.click(dropdownToggleBtn);
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+  });
+
+  it('does not open when a click happens anywhere aside from the dropdown toggle button'
+  + 'when the dropdown is closed', async function() {
+    const user = userEvent.setup(),
+        component = renderEl()!,
+        menu = component.querySelector('.nx-dropdown-menu');
+
+    expect(menu).not.toBeInTheDocument();
 
-    element!.unmount();
+    await user.click(document.body);
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('forwards a ref to the div', function() {
-    const ref = React.createRef<HTMLDivElement>(),
+  it('does not open if a click happens anywhere when the element is disabled', async function() {
+    const user = userEvent.setup(),
+        component = renderEl({ disabled: true })!,
+        dropdownToggleBtn = screen.getByRole('button', { name: 'more options' }),
+        menu = component.querySelector('.nx-dropdown-menu');
+
+    expect(menu).not.toBeInTheDocument();
+
+    await user.click(dropdownToggleBtn);
+    expect(menu).not.toBeInTheDocument();
+  });
 
-        // note: the fragment is necessary to get around an enzyme issue:
-        // https://github.com/enzymejs/enzyme/issues/1852#issuecomment-433145879
-        div = mount(<><NxStatefulSegmentedButton ref={ref} { ...minimalProps }/></>).children().children();
+  it('closes the dropdown when ESC is pressed on this component when dropdown is open', async function() {
+    const user = userEvent.setup(),
+        { container } = quickRender(),
+        dropdownToggleBtn = screen.getByRole('button', { name: 'more options' });
+    let menu = container.querySelector('.nx-dropdown-menu');
 
-    expect(ref.current).toBe(div.getDOMNode());
+    await user.click(dropdownToggleBtn);
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await fireEvent.keyDown(menu as HTMLElement, { key: 'Escape' });
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
   });
 
+  it('calls preventDefault on Escape keydown', async function() {
```

Comment:
Only when the dropdown is open

Reply:
> Only when the dropdown is open

[f25e053](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f25e053572e0e0069e299e9dbef87382cb8484b0)